thunder/transforms/materialization.py

```diff
@@ -66,9 +143,12 @@ def transform_module(self, model: ThunderModule):
                     b, device=getattr(b, "_thunder_device", self.device), requires_grad=b.requires_grad
                 )
                 for n2 in shared_names[n]:
-                    model._overrides_parameters[n2] = b
+                    model._overrides_buffers[n2] = b
                     self.have_materialized.add(n2)
 
+        if self.init != module_init_from_original_module_init:
```

Comment:
would this mean `module_init_from_original_module_init(..., non_persistent_only=True)` is called even when `init_from_original_state_dict` and `init_from_transformed_state_dict`?

Reply:
Yes, but that materializes only the submodules that contain non-persistent buffers.
I moved this to the `init_from_original_state_dict` and `init_from_transformed_state_dict` themselves and amended the comment to
```
# initialize things not persisted in the state dict
```